pkg/store/device/cache.go

```diff
@@ -142,6 +158,12 @@ func (c *networkChangeStoreCache) GetDevices() []*Info {
 	return devices
 }
 
+func (c *networkChangeStoreCache) Watch(ch chan<- *Info) {
```

Comment:
In this implementation, a `Watch` only sees devices that are added after the watch call. Controllers need to see all past and future devices to avoid race conditions resulting in missed events. The `Watch` method could instead look a lot like the `listen()` method does now. It would call `c.networkChangeStore.Watch(ch, networkchangestore.WithReplay())` and pass a new channel down to that call. Replaying the entire list of devices is really important for controllers to ensure events are not lost.

Also, we probably want to use the `stream.Context` that's used in other stores to avoid memory leaks:
https://github.com/onosproject/onos-config/blob/ed8b26d85b55cae8cfc485be311fe87e94908855/pkg/store/change/network/store.go#L135

This allows the `Watch` caller to close the stream when it stops listening. In controller watchers, when the watcher is stopped the stream is closed. Otherwise state changes in the cluster can result in the controllers continuously adding channels that are never removed.

Reply:
what I was concerned about is that watcher of this cache only wants the **updates** to the **cache** (which is only the device type and version) and not for instance to get an event every time a network change is modified. To do this it has to keep the current list of cache items, and only update it in one place - this is currently done in **listen**.

I get it that a replay is needed and the stream.Context - I will work on these, but I was reluctant to send up every network change